tracker/view/login.py

```diff
@@ -29,9 +35,36 @@ def login():
                                User=User,
                                password_length={'min': TRACKER_PASSWORD_LENGTH_MIN,
                                                 'max': TRACKER_PASSWORD_LENGTH_MAX}), status_code
-
+    if TRACKER_ENABLE_TWO_FACTOR:
+        session['two_factor'] = form.user.name
+        return redirect(url_for('tracker.two_factor'))
     user = user_assign_new_token(form.user)
-    user.is_authenticated = True
+    login_user(user)
+    return redirect(url_for('tracker.index'))
+
+
+@tracker.route('/twofactor', methods=['GET', 'POST'])
```

Comment:
I don't like this being a separate endpoint, all this logic should be in the login endpoint as there is no reason to have a separate accessible URL for this, its the second step of the same process: login